universe/interface.go

```diff
@@ -573,6 +573,19 @@ const (
 	ProofTypeTransfer
 )
 
+// NewProofTypeFromAssetProof returns the proof type for the given asset proof.
+func NewProofTypeFromAssetProof(proof *proof.Proof) (ProofType, error) {
+	if proof == nil {
+		return ProofTypeUnspecified, fmt.Errorf("proof is nil")
```

Comment:
micro-nit: for numerical values of a custom type we usually explicitly return `0` just to visually focus more on the error being non-nil.

Reply:
modified here: https://github.com/lightninglabs/taproot-assets/pull/540